### PR TITLE
feat(audio): internal test page for audio

### PR DESCRIPTION
**Pull request by andreasgangso**
## About the changes

Depends on https://github.com/bcc-code/bccm-flutter/pull/4

### CODE REVIEW

File: lib/api/bmm.dart [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+final bmmApiProvider = Provider<BmmApiWrapper>((ref) {
+  return BmmApiWrapper(
+    basePathOverride: 'https://bmm-api.brunstad.org',
+    getAuthToken: () => ref.read(authStateProvider).auth0AccessToken,
+    interceptors: [
+      InterceptorsWrapper(onRequest: (options, handler) {
+        options.headers['Authorization'] = 'Bearer ${ref.read(authStateProvider).auth0AccessToken}';
+        options.headers['Accept-Language'] = [ref.read(settingsProvider).appLanguage.languageCode];
+        options.headers['BMM-Version'] = '1.101.0';
```
Comment (sifferhans) [inline lib/api/bmm.dart:15]: Bør denne hard-kodes på denne måten?

File: lib/helpers/bmm/streak_listener.dart [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    final player = BccmPlayerController.primary;
+    onPlayerStateChanged() {
+      if (todaysFraKaareTrackId == null || eligibleUntil == null) return;
+      final extras = player.value.currentMediaItem?.metadata?.extras;
+      if (extras == null) return;
+      if (extras['type'] != 'track') return;
+      if (extras['id'] != todaysFraKaareTrackId) {
+        return;
+      }
+      if (player.value.playbackState == PlaybackState.playing) {}
```
Comment (sifferhans) [inline lib/helpers/bmm/streak_listener.dart:29]: Are we gonna do anything here?

File: lib/screens/audio/audio.dart [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+
+class BmmDiscoveryRenderer extends HookConsumerWidget {
+  const BmmDiscoveryRenderer({super.key, required this.documents});
+
+  final BuiltList<IAllDocumentModels> documents;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final groups = useMemoized(() {
+      // list of (bool isTileGroup, BuiltList<IAllDocumentModels> documents)
```
Comment (sifferhans) [inline lib/screens/audio/audio.dart:94]: This comment is incorrect, either remove or update it

**Review by sifferhans — CHANGES_REQUESTED** [review]

File: lib/api/bmm.dart (hunk 1 above)
Comment (andreasgangso) [reply lib/api/bmm.dart:15]: 🤷

Comment (andreasgangso) [reply lib/api/bmm.dart:15]: Maybe good to change it to something else after agreement from karsten but that can be done after merge

File: lib/helpers/bmm/streak_listener.dart (hunk 2 above)
Comment (andreasgangso) [reply lib/helpers/bmm/streak_listener.dart:29]: I just started on it, but decided to stop for now. So not in this PR.

**Review by sifferhans — DISMISSED** [review]

**Review by sifferhans — APPROVED** [review]

**Pull request merged**